### ISSUE TITLE
Transcoding is broken again

### ISSUE THREAD

**Issue by kyak** [gerbera/gerbera#2377]
Transcoding is broken once again (see #2341). This is not a very good sign. Either the code base is so fragile, or there is a lack of testing or attention to the transcoding feature.

The last known working release is 71b8827abd85fd7352040959bbf68b08855b4ecc. The 9d23b5b7415767bf5744418ccc93725db63dab3e is already not working. The latest master is not working as well.

I will attach the debug log shortly.

**Comment by whyman** [comment]
Hey @kyak, thanks for using master and making these reports, we appreciate it and it improves Gerbera for everyone.

**Comment by kyak** [comment]
Attached is the log file.

[gerb.log](https://github.com/gerbera/gerbera/files/7835518/gerb.log)

**Comment by whyman** [comment]
I can't see many possibly related commits, my guess would be https://github.com/gerbera/gerbera/commit/91dce72f62a2dc857b88b93e808854d12f884a2e, based only on the files changed

**Comment by kyak** [comment]
> I can't see many possibly related commits, my guess would be [91dce72](https://github.com/gerbera/gerbera/commit/91dce72f62a2dc857b88b93e808854d12f884a2e), based only on the files changed

There are not so many commits, just 10 of them. And the one you refer to is older than the one that actually works.

**Comment by whyman** [comment]
Hah, you are correct, apologies

**Comment by kyak** [comment]
This is the faulty commit: 1ec9b7ce64c5cf2bc81ae1449dd26c4a055e31d2. Its parent, 71b8827abd85fd7352040959bbf68b08855b4ecc, works fine as I mentioned.

**Comment by KarlStraussberger** [comment]
@kyak Yes, the old code is a total mess that's why we need to rework it with the risk of breaking stuff. In hope of some friendly folk to find the issues. Thanks

@whyman What about returning `-99` for transcoding https://github.com/gerbera/gerbera/blob/4ff062f408e0bb3acc3190a23ff16a11bc2bb6bc/src/file_request_handler.cc#L240 in order to make sure it is handled correctly

**Comment by whyman** [comment]
I tested that transcoding worked with that change though and indeed for mj profiles it's working in master. 

So I am wondering what edge caae I am missing here

**Comment by whyman** [comment]
So looking at your log it looks like transcoding is getting invoked correctly, and testing on master I get transcoded content.

There is also a change in there to move the generation of the DNLA headers (with the intention of them being applied to all resource requests), so I think that is likely the issue (esp as you are using a samsung device, which is strict about such things)

**Comment by kyak** [comment]
> So looking at your log it looks like transcoding is getting invoked correctly, and testing on master I get transcoded content.
> 
> There is also a change in that commit to move the generation of the DLNA headers (with the intention of them being applied to all resource requests), so I think that is likely the issue (esp as you are using a samsung device, which is strict about such things)

I've tried with another device (not samsung) and it doesn't trigger transcoding at all! I had the `client-flags` set in transcoding profile so that only the samsung device gets transcoded content (another device can consume original content just fine). I've removed the `client-flags` from transcoding profile, but what I see is that another device doesn't get transcoded content - no transcoding process is started and I can see that the content is original (for example, I can seek which I can't do with transcoded content).

Anyway, this could be a separate issue, and I'm now trying to build with your latest fix to see if it makes any difference.
